File: src/StudentManagerError3.java
import java.util.Scanner;

class Student {
    String id;
    String name;
    float score;
}

public class StudentManagerError3 {
    public static void addStudent(Student[] students, int n) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Nhap ma: ");
        students[n].id = sc.next();
        System.out.print("Nhap ten: ");
        students[n].name = sc.next();
        System.out.print("Nhap diem: ");
        students[n].score = sc.nextFloat();
        n++;
    }

    public static void showStudents(Student[] students, int n) {
        for (int i = 0; i < n; i++)
            System.out.println(students[i].id + " " + students[i].name + " " + students[i].score);
    }

    public static void main(String[] args) {
        Student[] students = new Student[10];
        int n = 0;
        int ch = 1;
        Scanner sc = new Scanner(System.in);
        while (ch != 0) {
            System.out.println("1. Them\n2. Hien thi\n0. Thoat");
            ch = sc.nextInt();
            if (ch == 1) addStudent(students, n);
            else if (ch == 2) showStudents(students, n);
        }
    }
}

File: src/student_manager_error3.c
#include <stdio.h>
#define MAX 10

struct Student {
    char id[10];
    char name[20];
    float score;
};

void addStudent(struct Student students[], int n) {
    printf("Nhap ma: ");
    gets(students[n].id);
    printf("Nhap ten: ");
    gets(students[n].name);
    printf("Nhap diem: ");
    scanf("%f", &students[n].score);
    n++;
}

void showStudents(struct Student students[], int n) {
    for (int i = 0; i < n; i++)
        printf("%s %s %f\n", students[i].id, students[i].name, students[i].score);
}

int main() {
    struct Student students[MAX];
    int n = 0;
    int ch = 1;
    while (ch != 0) {
        printf("1. Them\n2. Hien thi\n0. Thoat\n");
        scanf("%d", &ch);
        if (ch = 1) addStudent(students, n);
        else if (ch == 2) showStudents(students, n);
    }
    return 0;
}
